<launch>
	<arg name="node_start_delay" default="60.0"/>
	<arg name="nao_ip" default="$(optenv NAO_IP 10.0.1.230)" />
	<arg name="nao_port" default="$(optenv NAO_PORT 9559)" />
	<arg name="pepper_on" default="True"/>
	<arg name="pepper_camera_on" default="True"/>
	<arg name="mic_index" default="2" />
	<!-- Launch Chatbot -->
	<node pkg="shopping_mall_assistant" name="dialogue_server" type="dialogue_server.py" output="screen" />	
	<!-- Launch People and Face Detector -->
	<node pkg="shopping_mall_assistant" type="image_publisher_node.py" name="image_publisher_node" output="screen" args="--pepper_camera_on=$(arg pepper_camera_on) --ip=$(arg nao_ip) --port=$(arg nao_port)" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
	<node pkg="shopping_mall_assistant" type="face_detector_node.py" name="face_detector_node" output="screen" />
	<node pkg="shopping_mall_assistant" type="visualization_node.py" name="visualization_node" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
	<!-- Launch Audio Speech Recognition and Speech to Text -->
	<node pkg="shopping_mall_assistant" type="voice_detection_node.py" name="voice_detection_node" output="screen" args="--mic_index=$(arg mic_index)"/>
	<node pkg="shopping_mall_assistant" type="asr_node.py" name="asr_node" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
	<!-- Launch Pepper Nodes -->
	<group if='$(eval pepper_on==True)'>
		<!-- Pepper Nodes -->
		<node pkg="shopping_mall_assistant" type="wakeup_rest_node.py" name="wakeup_rest_node" required="true" args="--ip=$(arg nao_ip) --port=$(arg nao_port)" output="screen" />
		<node pkg="shopping_mall_assistant" type="animated_speech_node.py" name="animated_speech_node" required="true" args="--ip=$(arg nao_ip) --port=$(arg nao_port)" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
	</group>
	<!-- Launch Core Node -->
	<node pkg="shopping_mall_assistant" type="behavior_manager_node.py" name="behavior_manager_node" output="screen" args = "--pepper_on=$(arg pepper_on)" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
</launch>
